Remove py2neo dependency

ormsbee opened the pull request:
```
The py2neo package exists to support CourseGraph, which required the
Neo4j database. Support for it was removed before the Sumac cut, tracked
by the following DEPR:
  https://github.com/openedx/edx-platform/issues/34342

This dependency looks like a bit of leftover code that was overlooked
during the removal process.

This also upgrades other packages via "make upgrade".
```

feanil commented:
@ormsbee looks like there are issues with the `make upgrade` I'll take a look at that, but for now, you could remove py2neo and then just run `make compile-requirements` to not pull in the other upgrades on this PR.

ormsbee committed:
chore: remove py2neo as a dependency

The py2neo package exists to support CourseGraph, which required the
Neo4j database. Support for it was removed before the Sumac cut, tracked
by the following DEPR:
  https://github.com/openedx/edx-platform/issues/34342

It looks like the actual dependency install was just overlooked during
the removal process.

ormsbee commented:
@feanil: Thanks. That looks much cleaner anyhow.

bmtcril reviewed (approved):
👍 👍
